logrotate.c

```diff
@@ -2554,6 +2554,11 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    /* Creates the dir for state file and moves old
+       state file to new location */
+    mkdir(STATEDIR, 0755);
+    rename(OLDSTATEFILE, STATEFILE);
+
```

Comment:
I have just realized that this is a wrong and unexpected action to do in case the ```-s``` option is specified.  We should also check that we are not overwriting an existing state file, which will likely reintroduce the TOCTOU race I previously wanted to eliminate :-(